### ISSUE TITLE
misleading error report: [fatal] parse cmd flags err context deadline exceeded

### ISSUE THREAD

**Issue by andelf** [pingcap/pd#568]
```
/bin/pd-server --name="pd1" --client-urls="http://ip:2379" --peer-urls="http://ip:2380" --data-dir="/data/deploy/data.pd" --join=http://ip2:2379 --config=conf/pd.toml -L debug
# here after 10s
2017/03/10 19:26:15 main.go:43: [fatal] parse cmd flags err context deadline exceeded
```

and log file got nothing .

real reason is that ``--join`` addr is unreachable.

**Comment by disksing** [comment]
The reason is we try to connect other peers inside `Config.adjust()`. Seems already fixed by #564? @nolouch please verify it.

**Comment by nolouch** [comment]
yep, it has been fixed
